modules/dists/PrivateDist.chpl

```diff
@@ -116,6 +113,10 @@ class PrivateDom: BaseRectangularDom {
   proc dsiSetIndices(x: domain) { halt("cannot reassign private domain"); }
   proc dsiGetIndices() { return {0..numLocales-1}; }
 
+  proc dsiAssignDomain(rhs: domain, lhsPrivate:bool) {
+    halt("cannot reassign private domain");
```

Comment:
Not without some significant effort. autoCopy/initCopy for domains are resolved during resolution so they can be called later in callDestructors. If there were a compilation error here, any program using PrivateDist would fail with a resolution error.